plugin/storage/grpc/config/config.go

```diff
@@ -43,11 +51,61 @@ type ClientPluginServices struct {
 
 // PluginBuilder is used to create storage plugins. Implemented by Configuration.
 type PluginBuilder interface {
-	Build() (*ClientPluginServices, error)
+	Build(logger *zap.Logger) (*ClientPluginServices, error)
+	Close() error
 }
 
 // Build instantiates a PluginServices
-func (c *Configuration) Build() (*ClientPluginServices, error) {
+func (c *Configuration) Build(logger *zap.Logger) (*ClientPluginServices, error) {
+	if c.PluginBinary != "" {
+		return c.buildPlugin()
+	} else {
+		return c.buildRemote(logger)
+	}
+}
+
+func (c *Configuration) Close() error {
+	if c.PluginBinary == "" {
+		return c.RemoteTLS.Close()
+	}
+	return nil
```

Comment:
```suggestion
	return c.RemoteTLS.Close()
```
the check is not necessary, TLSConfig.Close() is safe to call